### PR TITLE
Feat/admin dashboard

### PR DESCRIPTION
**Pull request by cterrazas2**
### All Submissions:

- [X] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/pynecone-io/pynecone/blob/main/CONTRIBUTING.md) file?
- [X] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/pynecone-io/pynecone/pulls ) for the desired changed?

### Type of change

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

### New Feature Submission:

- [X] Does your submission pass the tests? 
- [X] Have you linted your code locally prior to submission?

### Changes To Core Features:

- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [X] Have you written new tests for your core changes, as applicable?
- [X] Have you successfully ran tests with your changes locally?

Description: Adds an **optional** admin dashboard to pynecone apps (auth not provided but shown below) using [starlette-admin](https://github.com/jowilf/starlette-admin). This is not required for previous versions.

Example Usage:

```python3
import pynecone as pc

class User(pc.Model, table=True):
    """A table for users in the database."""

    username: str
    password: str
"""

# Below is with a custom admin auth, see: https://jowilf.github.io/starlette-admin/tutorial/authentication/
custom_admin = Admin(
    engine,
    title="Example: Auth",
    base_url="/admin",
    statics_dir="examples/auth/static",
    login_logo_url="/admin/statics/logo.svg",  # base_url + '/statics/' + path_to_the_file
    auth_provider=MyAuthProvider(allow_paths=["/statics/logo.svg"]),
    middlewares=[Middleware(SessionMiddleware, secret_key=SECRET)],
)
# Then, just pass this custom Admin with Auth and whatever else you want to the Pynecone AdminDash
app = pc.App(state=State, admin_dash=pc.AdminDash(models=[User], admin=custom_admin))
"""
# Or with no custom admin/auth
app = pc.App(state=State, admin_dash=pc.AdminDash(models=[User]))
app.add_page(index)
app.compile()
```

Then in `pcconfig.py` make sure to set `enable_admin`:

```python3
import pynecone as pc


config = pc.Config(
    app_name="gpt",
    bun_path="$HOME/.bun/bin/bun",
    db_url="sqlite:///pynecone.db",
    enable_admin=True,
    env=pc.Env.DEV,
)
```

Then visit `localhost:8000/admin` to interact with all your models in the db with starlette-admin.

### CODE REVIEW

**Comment by Lendemor** [comment]
I think we shouldn't have to touch the code of the app to enable/disable admin dashboard.
It might be better if we could list the models we want in the admin Dashboard within the pcconfig.py files.

**Comment by cterrazas2** [comment]
@Lendemor so something like below?

```python3
config = pc.Config(
    app_name="gpt",
    bun_path="$HOME/.bun/bin/bun",
    db_url="sqlite:///pynecone.db",
    enable_admin=True,
    admin_dash=pc.AdminDash(
        models=[User, Question],  # Add/import models here, empty list if no models
        # admin=custom_admin,  # Add custom admin here, not required but auth not enabled without it
    ),
    env=pc.Env.DEV,
)
```

**Comment by masenf** [comment]
> Eventually we should replace this with a Pynecone UI with all the same features so we can customize it more.

@picklelo i've been working on a pynecone native admin interface for the last couple of weeks: https://github.com/trivial-intelligence/pynecone-admin

I saw this and sort of tabled my project because it would be a long time before mine has feature parity with the admin interface being integrated in this PR.

I'll probably keep developing mine (basically as an example) and split out the auth functionality, since that's probably useful in a wide array of apps.

**Comment by cterrazas2** [comment]
@picklelo this is ready to be merged, keep getting conflicts from main.

File: pynecone/config.py [hunk 1]
```diff
@@ -173,6 +174,12 @@ class Config(Base):
     # Additional frontend packages to install.
     frontend_packages: List[str] = []
 
+    # Enable the admin dash.
+    enable_admin: bool = False
+
+    # The Admin Dash
+    admin_dash: Optional[AdminDash] = None
```
Comment (picklelo) [inline pynecone/config.py:181]: I think we can combine these two fields into one. If the `admin_dash` is set to `None` we can disable it.

**Review by picklelo — APPROVED** [review]
awesome, thanks for adding this!

**Pull request merged**